apps/src/templates/WireframeButtons.jsx

```diff
@@ -0,0 +1,180 @@
+var React = require('react');
+
+var SendToPhone = window.dashboard ? window.dashboard.SendToPhone : undefined;
+
+var styles = {
+  main: {
+    font: '12pt "Gotham 5r", sans-serif',
+  },
+  sendToPhone: {
+    label: {
+      font: '12pt "Gotham 5r", sans-serif',
+      lineHeight: 'normal',
+      cursor: 'default',
+    },
+    div: {
+      margin: 0,
+      lineHeight: 'normal',
+    },
+  }
+};
+
+
+const PROJECT_URL_PATTERN = /^(.*\/projects\/\w+\/[\w\d-]+)\/.*/;
+
+
+/**
+ * List of app types for which we should show a "View code" button here. Other
+ * types will have a "How it works" button on the app itself, viewable on mobile
+ * devices.
+ */
+const APP_TYPES_WITH_VIEW_CODE = [
+  'applab',
+  'gamelab',
+  'makerlab',
+];
+
+
+/**
+ * Url to which the "Make my own" button should go to, based on the type of the
+ * currently displayed app, and whether or not it is the legacy version.
+ */
+const APP_TYPE_TO_NEW_PROJECT_URL = {
```

Comment:
I'd have to think about this a little more.  The JSDoc for `appToProjectUrl` is:
```
@returns {string} The path to the app capable of running this project as a standalone app.
```
I think it would be reasonable to augment that function to also handle legacy shares.  Thoughts?